memory/table.go

```diff
@@ -1415,6 +1415,7 @@ func (t *Table) ModifyColumn(ctx *sql.Context, columnName string, column *sql.Co
 		}
 	}
 
+	t.db.tables[t.name] = t
```

Comment:
This is supposed to happen when the session is committed. Why isn't it? You have to fix that bug, not mess up the global state here.